include/wil/resource.h

```diff
@@ -105,6 +105,7 @@ namespace wil
             typedef typename policy::pointer pointer;
             typedef unique_storage<policy> base_storage;
 
+        public:

```

Comment:
Chat indicates that this is required bc the "is nothrow constructable" check requires this type to be, well, constructable. Maybe to retain the whole "don't use this directly, use something else" retain the dtor as protected?

Reply:
As it turns out, if you make the destructor protected, `is_nothrow_constructible_v` doesn't work either. Both constructor and destructor have to be public, so that they can be used in the "forein", non-friend `is_nothrow_constructible` template.